Clock argument of run can be None

Trio's documentations explicitly states that the "clock" argument of `trio.run` can be `None`:

> - **clock** – `None` to use the default system-specific monotonic clock; otherwise, an object implementing the [`trio.abc.Clock`](https://trio.readthedocs.io/en/stable/reference-core.html#trio.abc.Clock) interface, like (for example) a [`trio.testing.MockClock`](https://trio.readthedocs.io/en/stable/reference-testing.html#trio.testing.MockClock) instance.

however the stub for it is showing only `trio.abc.Clock`:

https://github.com/python-trio/trio-typing/blob/9d57cb1d3f1fa7c9f915e7c884e9a2930645b8b7/trio-stubs/__init__.pyi#L166-L173

This is making it problematic to wrap `run` with a similar calling interface:

```python
from trio import run
from trio.abc import Clock

def my_run(clock: Clock|None = None) -> None:
    run(my_async_fn, clock=clock)

async def my_async_fn() -> None:
   ...
```

```sh
$ mypy sample.py
sample.py:5: error: Argument "clock" to "run" has incompatible type "Optional[Clock]"; expected "Clock"
Found 1 error in 1 file (checked 1 source file)
```

--- a/trio-stubs/__init__.pyi
+++ b/trio-stubs/__init__.pyi
@@ -167,7 +167,7 @@ def current_time() -> float: ...
 def run(
     afn: Union[Callable[..., Awaitable[T]], Callable[[VarArg()], Awaitable[T]]],
     *args: Any,
-    clock: trio.abc.Clock = ...,
+    clock: Optional[trio.abc.Clock] = ...,
     instruments: Sequence[trio.abc.Instrument] = ...,
     restrict_keyboard_interrupt_to_checkpoints: bool = ...,
 ) -> T: ...

--- a/trio-stubs/lowlevel.pyi
+++ b/trio-stubs/lowlevel.pyi
@@ -95,7 +95,7 @@ def start_guest_run(
     done_callback: Callable[[outcome.Outcome[T]], None],
     run_sync_soon_not_threadsafe: Callable[[Callable[[], None]], None] = ...,
     host_uses_signal_set_wakeup_fd: bool = ...,
-    clock: trio.abc.Clock = ...,
+    clock: Optional[trio.abc.Clock] = ...,
     instruments: Sequence[trio.abc.Instrument] = ...,
     restrict_keyboard_interrupt_to_checkpoints: bool = ...,
 ) -> None: ...

